### ISSUE TITLE
[stencil] CRUD APIs out of the box

### ISSUE THREAD

**Issue by techsavvyash** [SamagraX-Stencil/stencil#2]
## ToDo: 

- [ ] Add support for having support for CRUD APIs out of box. 

## References: 
- https://docs.nestjs.com/recipes/crud-generator
- https://github.com/nestjsx/crud

**Comment by bandevedant** [comment]
@RyanWalker277 @techsavvyash we have to just write the api's in typescript cna you please say me the file structure of the folder ...or can u please explain me what is expected .Thank you.

**Comment by RyanWalker277** [comment]
> @RyanWalker277 @techsavvyash we have to just write the api's in typescript can you please say me the file structure of the folder ...or can u please explain me what is expected .Thank you.

Hey @bandevedant, you can take reference from here: https://geekiebarbs.hashnode.dev/build-a-crud-api-with-typescript-nodejs-express-and-postgresql

We need to provide sample CRUD APIs when a new project is initialised. The samples/01-all-features is a nest project. You have to make your contributions there.

**Comment by bandevedant** [comment]
Okay please assign me this issue .

**Comment by Sandeep0306** [comment]
Hi, @RyanWalker277 is someone working on it? if not can you assign it to me?

**Comment by RyanWalker277** [comment]
@bandevedant is there any update on the progress?

**Comment by Sandeep0306** [comment]
@RyanWalker277 bro assign it to me ?

**Comment by bandevedant** [comment]
yes @Sandeep0306  go for it I am in with some other stuff

**Comment by ALOK9442** [comment]
> > @RyanWalker277 @techsavvyash we have to just write the api's in typescript can you please say me the file structure of the folder ...or can u please explain me what is expected .Thank you.
> 
> Hey @bandevedant, you can take reference from here: https://geekiebarbs.hashnode.dev/build-a-crud-api-with-typescript-nodejs-express-and-postgresql
> 
> We need to provide sample CRUD APIs when a new project is initialised. The samples/01-all-features is a nest project. You have to make your contributions there.

@RyanWalker277 hello sir, can you tell more about this, I am confused about exactly what customization required?

**Comment by techsavvyash** [comment]
Hey @ALOK9442 
Refer this: https://github.com/nestjsx/crud

Since we are using Prisma, we need something similar which works with Prisma.

**Comment by ALOK9442** [comment]
we have to use nest not express?

**Comment by techsavvyash** [comment]
Yes, Stencil is based on Nest not express. The main aim of this ticket to help enable CRUD APIs in a single command for schema models so that the developers don't have to write the same logic again and again.

**Comment by Apurv428** [comment]
> > @RyanWalker277 @techsavvyash we have to just write the api's in typescript can you please say me the file structure of the folder ...or can u please explain me what is expected .Thank you.
> 
> Hey @bandevedant, you can take reference from here: https://geekiebarbs.hashnode.dev/build-a-crud-api-with-typescript-nodejs-express-and-postgresql
> 
> We need to provide sample CRUD APIs when a new project is initialised. The samples/01-all-features is a nest project. You have to make your contributions there.

Hi @RyanWalker277 , where can I find samples/01-all-features project? I am unable to get a link for that.

**Comment by techsavvyash** [comment]
@Apurv428 its in the root of this project.

**Comment by Apurv428** [comment]
@techsavvyash can you please provide the sample template for the prisma schema?

**Comment by Apurv428** [comment]
@techsavvyash @RyanWalker277 
I have created the APIs using NestJS. I have a doubt that can there be multiple models in the schema.prisma?
If there are multiple models, then can we have the api link as
`http://localhost:3000/api/v1/model1 http://localhost:3000/api/v1/model2` etc?

**Comment by Apurv428** [comment]
Okay, thanks! I will make the PR after making the changes. Can you please assign this to me?

**Comment by Apurv428** [comment]
Hi @techsavvyash , do we need a file in which the models will be stored after retrieving from schema.prisma?

For eg in schema.prisma, the model is as follow:

```
model Book {
  id Int @id @default(autoincrement())
  title String
  description String?
}
```
Then do we need to create a file in which the model that is retrived is stored in the format understandable by the api?

```
export class Book implements Prisma.BookCreateInput {
  id: number;
  title: string;
  description?: string;
}
```

**Comment by techsavvyash** [comment]
Yes @Apurv428,
It would be good to have DTOs and Interfaces generated as well.

**Comment by Apurv428** [comment]
Okay, thanks!

**Comment by Yash-Sajwan24** [comment]
@techsavvyash  I understand the problem, and I believe I have the necessary skills. Could I have the opportunity to work on this?

**Comment by aryangandhi05** [comment]
Hello @techsavvyash Sir I want to solve this issue, as i understand it, and i will give my 100% for solving this issue So please, can you assign me this?

**Comment by Apurv428** [comment]
@Yash-Sajwan24 @aryangandhi05 , I am working on it already!

**Comment by techsavvyash** [comment]
@Yash-Sajwan24 @aryangandhi05 the first version of this extension has been very graciously contributed by @Apurv428 and he is continuing to work on this ticket to improve the extension. I suggest the three of you team up and work together, might help expedite the process.

**Comment by Yash-Sajwan24** [comment]
@techsavvyash  @Apurv428  Yes, I would love to team up.

**Comment by Paras-codes** [comment]
hii @techsavvyash  can i also team up  with apurv as I have skills in API development and would really like to build this feature

**Comment by jayesnc2512** [comment]
Hi @RyanWalker277 Hope you're doing.
I Have a Doubt regarding making an API.
Using which db platform are you looking for. 
Or does it need to have generalized for now.

**Comment by Savio629** [comment]
Any updates @Apurv428 ?

**Comment by Apurv428** [comment]
Sorry, due to my exams I was unable to work on it. I would continue the work and would create a PR asap!

**Comment by DarrenDsouza7273** [comment]
.

**Comment by techsavvyash** [comment]
## Resources

- [NestJS-X/CRUD Decorator Wiki](https://github.com/nestjsx/crud/wiki/Controllers#description)

**Comment by Srijan25** [comment]
is this issue resolved or not ? or should i give it a try

**Comment by Savio629** [comment]
@Srijan25 I'm currently working on this

**Comment by VedantKhairnar** [comment]
Hello @Savio629 
Any update on the contribution?

**Comment by swatiiyadawar** [comment]
@techsavvyash  Hi sir, I'm 100% capable of doing this and I will love to contribute so please assign it to me

**Comment by prince04-coder** [comment]
@techsavvyash sir, i can resolve this issue, so please assign it to me

**Comment by techsavvyash** [comment]
@prince04-coder please go ahead and raise a draft PR assignment only comes once there is a draft PR so that we know that you have an action bias and will actually work on the issue.
